### PR TITLE
Add and improve examples

### PR DESCRIPTION
**Pull request by akaszynski**
With #1457, we can start to improve our documentation examples.  Combined with isolated documentation pages for each method, we can have examples match how `numpy`, `scipy`, and `pandas` create their examples.

tmp

### CODE REVIEW

File: pyvista/plotting/renderer.py [hunk 1]
```diff
@@ -505,15 +578,55 @@ def add_axes(self, interactive=None, line_width=2,
...
             The width of the marker lines
 
-        box : bool
+        box : bool, optional
             Show a box orientation marker. Use ``box_args`` to adjust.
             See :any:`pyvista.create_axes_orientation_box` for details.
 
-        opacity : int or float, optional
-            The opacity of the marker.
+        xcolor : str, optional
```
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:588]: x_color, y_color, z_color.  Should we put the default behavior?

File: pyvista/plotting/renderer.py [hunk 2]
```diff
@@ -1311,6 +1617,31 @@ def set_scale(self, xscale=None, yscale=None, zscale=None, reset_camera=True):
         Scaling in performed independently on the X, Y and Z axis.
         A scale of zero is illegal and will be replaced with one.
 
+        Parameters
+        ----------
+        xscale : float, optional
+            Scaling in the x direction.
```
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:1623]: The default behavior is unclear.

File: pyvista/plotting/renderer.py [hunk 3]
```diff
@@ -1389,22 +1732,36 @@ def reset_camera(self, render=True, bounds=None):
             self.parent.render()
         self.Modified()
 
-    def isometric_view(self):
+    def view_isometric(self, negative=False):
         """Reset the camera to a default isometric view.
 
-        DEPRECATED: Please use ``view_isometric``.
```
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:1395]: Similar to above, this PR is also removing functionality that is deprecated.  It may be best in a separate PR.

File: pyvista/plotting/renderer.py [hunk 4]
```diff
@@ -1533,31 +1948,49 @@ def disable_shadows(self):
...
+        Examples
+        --------
+        Set the background color to black with a gradient to white at
+        the top of the plot.
+
+        >>> import pyvista
+        >>> pl = pyvista.Plotter()
+        >>> actor = pl.add_mesh(pyvista.Cone())
+        >>> pl.set_background('black', top='white')
+        >>> pl.show()
```
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:1993]: Need a newline here to be consistent

File: pyvista/utilities/errors.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (MatthewFlamm) [inline pyvista/utilities/errors.py:1]:
```suggestion
```

File: pyvista/plotting/plotting.py [hunk 6]
```diff
@@ -573,10 +637,10 @@ def reset_camera(self, *args, **kwargs):
...
         self.render()
 
-    @wraps(Renderer.isometric_view)
-    def isometric_view(self, *args, **kwargs):
-        """Wrap ``Renderer.isometric_view``."""
-        return self.renderer.isometric_view(*args, **kwargs)
+    def isometric_view(self):
+        """Please use ``view_isometric``, this method has been deprecated."""
+        from pyvista.core.errors import DeprecationError
+        raise DeprecationError('Please use ``view_isometric``.')
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/plotting/plotting.py:643]: > Is there a reason to include deprecations in this PR? This seems like something best done in a separate PR to keep the history understandable.

No reason.  I tend to do this with my PRs and it's bad form.  Will revert.

File: pyvista/plotting/renderer.py (hunk 1 above)
Comment (akaszynski) [reply pyvista/plotting/renderer.py:588]: The default is seen in the docstring.  We've established that it's optional to provide the defaults in our docstrings.

File: pyvista/plotting/renderer.py [hunk 7]
```diff
@@ -541,13 +654,29 @@ def add_axes(self, interactive=None, line_width=2,
...
 
     def show_axes(self):
-        """Show the axes orientation widget."""
+        """Show the axes orientation widget.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> pl = pyvista.Plotter()
+        >>> pl.hide_axes()
```
Comment (akaszynski) [inline pyvista/plotting/renderer.py:677]:
```suggestion
        >>> pl.show_axes()
```

File: pyvista/plotting/renderer.py [hunk 8]
```diff
@@ -433,10 +455,46 @@ def add_axes_at_origin(self, x_color=None, y_color=None, z_color=None,
...
+            The color of the z axes arror.
+
+        xlabel : str, optional
+            The label of the x axes arror.
+
+        ylabel : str, optional
+            The label of the y axes arror.
+
+        zlabel : str, optional
+            The label of the z axes arror.
```
Comment (akaszynski) [inline pyvista/plotting/renderer.py:476]:
```suggestion
            The color of the x axes arrow.

        y_color : string or 3 item sequence, optional
            The color of the y axes arrow.

        z_color : string or 3 item sequence, optional
            The color of the z axes arrow.

        xlabel : str, optional
            The label of the x axes arrow.

        ylabel : str, optional
            The label of the y axes arrow.

        zlabel : str, optional
            The label of the z axes arrow.
```

File: pyvista/plotting/renderer.py [hunk 9]
```diff
@@ -433,10 +455,46 @@ def add_axes_at_origin(self, x_color=None, y_color=None, z_color=None,
                            labels_off=False):
         """Add axes actor at origin.
 
+        Parameters
+        ----------
+        x_color : string or 3 item sequence, optional
+            The color of the x axes arror.
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/plotting/renderer.py:461]: Good catch.  Correcting.

File: pyvista/plotting/renderer.py [hunk 10]
```diff
@@ -688,7 +827,8 @@ def show_bounds(self, mesh=None, bounds=None, show_xaxis=True,
         >>> mesh = pyvista.Sphere()
         >>> plotter = pyvista.Plotter()
         >>> actor = plotter.add_mesh(mesh)
-        >>> actor = plotter.show_bounds(grid='front', location='outer', all_edges=True)
+        >>> actor = plotter.show_bounds(grid='front', location='outer', 
+        ...                             all_edges=True)
```
[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/plotting/renderer.py:831]: I'll do a rgrep for these and fix it.

File: pyvista/plotting/renderer.py [hunk 11]
```diff
@@ -688,7 +827,8 @@ def show_bounds(self, mesh=None, bounds=None, show_xaxis=True,
...
         >>> cpos = plotter.show()
```
Comment (akaszynski) [inline pyvista/plotting/renderer.py:832]:
```suggestion
        >>> plotter.show()
```

File: pyvista/plotting/renderer.py [hunk 12]
```diff
@@ -1311,6 +1617,31 @@ def set_scale(self, xscale=None, yscale=None, zscale=None, reset_camera=True):
...
+        Parameters
+        ----------
+        xscale : float, optional
+            Scaling in the x direction.
+
+        yscale : float, optional
+            Scaling in the y direction.
+
+        zscale : float, optional
+            Scaling in the z direction.
```
Comment (akaszynski) [inline pyvista/plotting/renderer.py:1629]:
```suggestion
            Scaling in the x direction.  Default is ``None``, which does not change existing scaling.

        yscale : float, optional
            Scaling in the y direction.  Default is ``None``, which does not change existing scaling.

        zscale : float, optional
            Scaling in the z direction.  Default is ``None``, which does not change existing scaling.
```

File: pyvista/plotting/renderer.py (hunk 2 above)
Comment (akaszynski) [reply pyvista/plotting/renderer.py:1623]: Clarified.

File: pyvista/plotting/renderer.py (hunk 4 above)
Comment (akaszynski) [inline pyvista/plotting/renderer.py:1993]:
```suggestion
        >>> pl.show()

```

File: pyvista/plotting/renderer.py (hunk 1 above)
Comment (MatthewFlamm) [inline pyvista/plotting/renderer.py:588]: These look unfixed (x_color vs xcolor), maybe the commits aren't pushed yet.  But I think the description should be something like "Color of x axis label", which I missed before

Comment (akaszynski) [reply pyvista/plotting/renderer.py]: Fixed in https://github.com/pyvista/pyvista/pull/1479/commits/6a74d3cf1f3a5c230a154ec78b4178f713d33588

**Comment by akaszynski** [comment]
@MatthewFlamm, I'm going to merge this as some of the changes here are needed by #1478 (i.e. removing cpos).  If there's more work to be done, please ping me and I'll create a PR.
